arrow-core-data/src/main/kotlin/arrow/core/Eval.kt

```diff
@@ -183,7 +183,7 @@ sealed class Eval<out A> : EvalOf<A> {
     fun raise(t: Throwable): Eval<Nothing> =
       defer { throw t }
 
-    val unit: Eval<Unit> = Now(kotlin.Unit)
+    val Unit: Eval<Unit> = Now(kotlin.Unit)
```

Comment:
This was uppercased on `0.11.0`

Reply:
Do we want those to be upper-cased or lowercased in Arrow for other libs and API's?

Reply:
Looking at the other (smart) constructors in Arrow, I think we should deprecate all of these in favor of their right hand side implementations. 

```
val Unit: Eval<Unit> = Now(kotlin.Unit)
val True: Eval<Boolean> = Now(true)
val False: Eval<Boolean> = Now(false)
val Zero: Eval<Int> = Now(0)
val One: Eval<Int> = Now(1)
```

If all agree I can add this change in this PR as well.

Reply:
I am sorry, I saw the reply after merging it. 
Let's move that to a different PR 😄